include/nil/blueprint/input_reader.hpp

```diff
@@ -164,50 +209,69 @@ namespace nil {
                     return res;
                     break;
                 default:
+                    error << "Expected int or string as a field value, got \"" << value << "\"";
                     return {};
                 }
             }
 
-            std::vector<var> process_field (llvm::GaloisFieldType *field_type, const boost::json::object &value, bool is_private) {
-                ASSERT(llvm::isa<llvm::GaloisFieldType>(field_type));
-
-                size_t arg_len = field_arg_num<BlueprintFieldType>(field_type);
-                ASSERT_MSG(arg_len != 0, "wrong input size");
+            bool check_field(const signature_node &node, llvm::GaloisFieldKind kind) {
+                if (node.elem != json_elem::FIELD) {
+                    return false;
+                }
+                if (node.children.size() > 1) {
+                    return false;
+                }
+                if (node.children.size() == 1) {
+                    auto corresponding_elem = [](llvm::GaloisFieldKind kind) {
+                        switch (kind) {
+                            case llvm::GALOIS_FIELD_PALLAS_BASE:
+                                return json_elem::PALLAS_BASE;
```

Comment:
implicit conversion from `json_elem` to `bool`

Reply:
lambda again